src/core/public/http/fetch.ts

```diff
@@ -95,34 +96,29 @@ export class Fetch {
     });
   };
 
-  private createRequest(path: string, options?: HttpFetchOptions): Request {
+  private createRequest(path: string, options: HttpFetchOptions = {}): Request {
     // Merge and destructure options out that are not applicable to the Fetch API.
     const { query, prependBasePath: shouldPrependBasePath, asResponse, ...fetchOptions } = merge(
       {
         method: 'GET',
         credentials: 'same-origin',
         prependBasePath: true,
+      },
+      options,
+      {
         headers: {
-          'kbn-version': this.params.kibanaVersion,
           'Content-Type': 'application/json',
+          [KbnHeaders.VERSION]: this.params.kibanaVersion,
+          [KbnHeaders.SYSTEM_API]: options.asSystemApi ? 'true' : undefined,
+          ...options.headers,
```

Comment:
Plugins can rewrite internal Kibana headers. Is it desired behaviour?

Reply:
Nope, added test and fixed.